webdriver/tests/bidi/browsing_context/navigate/fragment_navigated.py

```diff
@@ -0,0 +1,95 @@
+import pytest
+
+from tests.support.sync import AsyncPoll
+from webdriver.bidi.modules.script import ContextTarget
+
+from ... import any_int, any_string, recursive_compare
+
+pytestmark = pytest.mark.asyncio
+
+PAGE_EMPTY = "/webdriver/tests/bidi/browsing_context/navigate/support/empty.html"
+FRAGMENT_NAVIGATED_EVENT = "browsingContext.fragmentNavigated"
+
+
+transitions = [
+    ("", "#foo"),
+    ("#foo", "#bar"),
+    ("#foo", "#foo"),
+    ("#bar", ""),
+]
+
+transition_ids = [
+    "without hash to with hash",
+    "with different hashes",
+    "with identical hashes",
+    "with hash to without hash",
+]
+
+
+@pytest.mark.parametrize(
+    "hash_before, hash_after",
+    transitions,
+    ids=transition_ids,
+)
+async def test_fragment_navigated_via_navigate(
```

Comment:
should I perhaps parametrize further? will probably do that tomorrow

Reply:
I mean smth like

```
@pytest.mark.parametrize(
    "navigationType",
   ['browsing_context.navigate', 'script.call_function'],
    ids=['BiDi navigation', 'script evaluation'],
)
```

Reply:
Yes!